test/controllers/architectures_controller_test.rb

```diff
@@ -80,4 +80,17 @@ def user_with_viewer_rights_should_succeed_in_viewing_architectures
     get :index
     assert_response :success
   end
+
+  # randomly selected architectures controller for the following test
+  def should_render_pagination_when_needed
+    old = Setting[:entries_per_page]
+    Setting[:entries_per_page] = Architecture.count
+    get :index, {}, set_session_user
+    refute_includes @response.body, 'pagination'
+    FactoryGirl.create(:architecture)
+    get :index, {}, set_session_user
+    assert_includes @response.body, 'pagination'
+  ensure
+    Setting[:entries_per_page] = old
+  end
```

Comment:
I'm missing tests to the following:
- per_page dropdown exists with the correct values
- selecting per_page redirect to the correct page and has the correct number of entries
- first / next / prev / last links exists
- controller respects the per_page and page params